src/modules/verifiable-credentials/verifiable-credentials-base.service.ts

```diff
@@ -93,12 +101,43 @@ export abstract class VerifiableCredentialsServiceBase {
     proof_options: string
   ): Promise<string>;
 
-  constructor(protected readonly _signerService: SignerService) {}
+  constructor(
+    protected readonly _signerService: SignerService,
+    private readonly _storage: VCStorageClient
+  ) {}
+
+  /**
+   * @description requests verifiable credentials of type `type`. Only EWF credentials currently supported
+   * @returns credentials query with matching verifiable presentations
+   */
+  async initiateExchange({ type, url }: ExchangeInvitation) {
+    if (type !== EWF_EXCHANGE_INVITATION) {
+      throw new Error('Only EWF exchange is supported');
+    }
+
+    const { data: vpRequest } = await axios.post<VpRequest>(url);
+    const presentationDefinitions = vpRequest.query.find(
+      (q) => q.type === VpRequestQueryType.presentationDefinition
+    )?.credentialQuery as PresentationDefinition[];
+
+    const vc: VerifiableCredential<RoleCredentialSubject>[] = (
+      await Promise.all(
+        presentationDefinitions.map((def) =>
+          this._storage.getCredentialsByDefinition(def)
+        )
+      )
+    )
+      .reduce((prev, curr) => prev.concat(curr), [])
+      .filter(isRoleCredential);
```

Comment:
I don't think restricting this to `RoleCredentials` is necessary, unless I'm missing something
```suggestion
    const vc: VerifiableCredential<CredentialSubject>[] = (
      await Promise.all(
        presentationDefinitions.map((def) =>
          this._storage.getCredentialsByDefinition(def)
        )
      )
    )
      .reduce((prev, curr) => prev.concat(curr), []);
```

Reply:
This is because we create presentation of role credentials https://github.com/energywebfoundation/iam-client-lib/blob/b96d23a4c0ed021a7c229ea39ecfd4b03cca0849/src/modules/verifiable-credentials/verifiable-credentials-base.service.ts#L197